core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -288,9 +288,7 @@ static void addBindPermissions(Permissions policy, Settings settings) throws IOE
         for (Map.Entry<String, Settings> entry : profiles.entrySet()) {
             Settings profileSettings = entry.getValue();
             String name = entry.getKey();
-            String transportRange = profileSettings.get("port",
-                                        settings.get("transport.tcp.port",
-                                                NettyTransport.DEFAULT_PORT_RANGE));
+            String transportRange = profileSettings.get("port",NettyTransport.PORT.get(settings));
```

Comment:
I agree it is a real problem. But i think the proper solution is to refactor the netty support into a module, that is really needed.